pynopegl-utils/pynopegl_utils/qml/viewer.qml

```diff
@@ -0,0 +1,503 @@
+import QtQuick.Dialogs
+import QtQuick.Layouts
+import QtQuick.Templates as T
+import Qt.labs.qmlmodels // DelegateChooser is experimental
+
+ApplicationWindow {
+    visible: true
+    minimumWidth: 640
+    minimumHeight: 480
+
+    signal exportVideo(string filename, int res, int profile, int samples)
+    signal cancelExport()
+
+    FileDialog {
+        id: scriptDialog
+        objectName: "scriptDialog"
+        nameFilters: ["Python scripts (*.py)"]
+        onAccepted: {
+            script.text = selectedFile
+            script.editingFinished()
+        }
+    }
+
+
+    /*
+     * Export
+     */
+
+    FileDialog {
+        id: exportDialog
+        objectName: "exportDialog"
+        nameFilters: ["MP4 (*.mp4)", "Animated GIF (*.gif)"]
+        fileMode: FileDialog.SaveFile
+        onAccepted: {
+            exportFile.text = selectedFile
+            exportFile.editingFinished()
+        }
+    }
+
+    Popup {
+        id: exportPopup
+
+        modal: true
+        closePolicy: Popup.NoAutoClose
+
+        anchors.centerIn: parent
+        StackLayout {
+            id: exportStack
+
+            // 0: Exporting
+            ColumnLayout {
+                Label { text: "Exporting " + exportFile.text }
+                ProgressBar {
+                    objectName: "exportBar"
+                    id: exportBar
+                    Layout.fillWidth: true
+                }
+                Button {
+                    text: "Cancel"
+                    Layout.alignment: Qt.AlignHCenter
+                    onClicked: {
+                        cancelExport()
+                        exportPopup.visible = false
+                    }
+                }
+            }
+
+            // 1: Export succeed
+            ColumnLayout {
+                Item { Layout.fillWidth: true; Layout.fillHeight: true } // spacer
+                Label { text: exportFile.text + " export done." }
+                Item { Layout.fillWidth: true; Layout.fillHeight: true } // spacer
+                Button {
+                    text: "OK"
+                    Layout.alignment: Qt.AlignHCenter
+                    onClicked: exportPopup.visible = false
+                }
+            }
+
+            // 2: Export failed
+            ColumnLayout {
+                Label { text: exportFile.text + " export failed."; color: "red" }
+                Label { id: exportErrorText }
+                Item { Layout.fillWidth: true; Layout.fillHeight: true } // spacer
+                Button {
+                    text: "OK"
+                    Layout.alignment: Qt.AlignHCenter
+                    onClicked: exportPopup.visible = false
+                }
+            }
+        }
+    }
+
+    function start_export(filename, res, profile, samples) {
+        exportBar.value = 0
+        exportStack.currentIndex = 0
+        exportPopup.visible = true
+        exportVideo(filename, res, profile, samples)
+    }
+
+    function done_export() {
```

Comment:
`done_export()` -> `finish_export()` ? (so we have start/finish)

Reply:
Renamed